packages/flutter_tools/lib/src/commands/precache.dart

```diff
@@ -111,6 +111,11 @@ class PrecacheCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    // Re-lock the cache.
+    if (globals.platform.environment['FLUTTER_ALREADY_LOCKED'] != 'true') {
+      await Cache.lock();
```

Comment:
Pre-cache still needs the cache lock.  I tested this manually and confirmed two concurrent `precache` commands wait for each other.

Reply:
Could you add a command test for the re-lock? somehow..

Reply:
Added two new tests.  Kind of lame but I needed:
```
  @visibleForTesting
  static bool isLocked() {
    return _lock != null;
  }
```